### ISSUE TITLE
build props from sentence objects

### ISSUE THREAD

**Issue by benbrastmckie** [benbrastmckie/ModelChecker#50]
I've been trying to make instances of `Sentence` (call these sentence objects) the objects from which propositions are built. The hope is to help consolidate things so that sentence objects can be used for everything, calling their different attributes as need be in different contexts. I think it is somewhat close, but there is a bug I can't seem to find. Here is the basic idea:

1. An infix sentence is used to generate a sentence object, including attributes for the main operator (type string) and the arguments (type sentence objects) in addition to what we had before.
2. There is also an attribute in each sentence object for the `prefix_sentence` or what I think might be better to call `prefix_obj` which is the instance of `prefix_type`. This attribute gets updated in `model_constraints` via the `instantiate` method. Before only the premises and conclusions were getting updated, but I tried to add the same for intermediate and sentence letters. I think the bug may be in the way the sentence letters are getting updated since it seems to find 'None' when looking for the `prefix_sentence` (`prefix_obj`) for sentence letters, so sentence letters seem like they are not getting updates.
3. I also added an attribute for `proposition_obj` with the hope that in `model_structure`, propositions could be found and added to sentence objects instead of added to a dictionary. That way we only need to keep track of the sentence objects.
4. Instead of maintaining a list of sentence objects for the premises and conclusions in syntax or elsewhere, the aim was to have a dictionary of all sentence objects where we can draw on the originally stored lists of infix premises and conclusions in order to look up and print whatever we want from a given sentences.

I added some inline comments as well. Hopefully that sorta makes sense. I feel like these foundational issues are important to settle since they may help to streamline everything (or that is the hope) at least conceptually. Curious to get your thoughts about these issues. Seems like we are close to straightening out the changes that have come from introducing the `Sentence` class and expanding its application throughout the project.
    -

**Comment by benbrastmckie** [comment]
I added a bunch of comments to `syntactic.py`. Seems like things will be able to be simplified a lot by defining whatever we will end up needing as soon as possible in `Sentence` and then pulling from there. I think I got an attribute for `letter_type` working that way, but still haven't been able to confirm since I still can't find where the bug is. I added a HINT where I think it is happening.
